navis/nbl/smat.py

```diff
@@ -247,37 +268,55 @@ def _query_many(self, idx_pairs, threads=None) -> Iterator[List[np.ndarray]]:
         chunks = chunksize(len(idx_pairs), threads)
 
         with ProcessPoolExecutor(threads) as exe:
-            yield from exe.map(
-                self._query, idx_pairs[:, 0], idx_pairs[:, 1], chunksize=chunks
-            )
+            yield from exe.map(self.match_fn,
+                               [self.objects[ix] for ix in idx_pairs[:, 0]],
+                               [self.objects[ix] for ix in idx_pairs[:, 1]],
+                               chunksize=chunks)
 
     def _query_to_idxs(self, q_idx, t_idx, counts=None):
-        """Produce a digitized counts array from a given query-target pair"""
+        """Produce a digitized counts array from a given query-target pair."""
         return self._count_results(self._query(q_idx, t_idx), counts)
 
     def _count_results(self, results: List[np.ndarray], counts=None):
         """Convert raw match function ouput into a digitized counts array.
 
         Requires digitizers.
         """
+        # Digitize
         idxs = [dig(r) for dig, r in zip(self.digitizers, results)]
 
+        # Make a stack
```

Comment:
It's faster to use numpy.unique to get the counts per cell than to tot up individual values.